storage/rocksdb/ha_rocksdb.cc

```diff
@@ -292,6 +292,42 @@ static int rocksdb_force_flush_memtable_now(
   return HA_EXIT_SUCCESS;
 }
 
+static void rocksdb_force_flush_memtable_and_lzero_now_stub(
+    THD *const thd, struct st_mysql_sys_var *const var, void *const var_ptr,
+    const void *const save) {}
+
+static int rocksdb_force_flush_memtable_and_lzero_now(
+    THD *const thd, struct st_mysql_sys_var *const var, void *const var_ptr,
+    struct st_mysql_value *const value) {
+  sql_print_information("RocksDB: Manual memtable and L0 flush\n");
+  rocksdb_flush_all_memtables();
+
+  const Rdb_cf_manager &cf_manager = rdb_get_cf_manager();
+  for (const auto &cf_handle : cf_manager.get_all_cf()) {
+    rocksdb::ColumnFamilyMetaData metadata;
+    rdb->GetColumnFamilyMetaData(cf_handle, &metadata);
```

Comment:
When calling into GetColumnFamilyMetaData() the current SuperVersion is referenced and the metadata is copied over from that. When the function exits the SuperVersion is de-referenced. A SuperVersion represents an immutable state of the database (when the manifest is updated). It is possible, however, that we are not the first to request that a given L0 file be compacted. This would cause CompactFiles to error out.

It is also possible that by the time we request to have a file flushed the file has been deleted because we have de-referenced the SuperVersion from which the metadata was copied. To solve this we should hold our own superversion reference and copy over the metadata ourselves.

This still doesn't seem like an issue to me considering this is to be used for performance testing/debugging. There is no copy-race condition and CompactFiles() should gracefully handle all sanity checks.